packages/@react-spectrum/dialog/test/AlertDialog.test.js

```diff
@@ -177,4 +213,114 @@ describe('AlertDialog', function () {
     let button = getByText('secondary').closest('button');
     expect(document.activeElement).toBe(button);
   });
+
+  it('render the primary button as pending', async function () {
+    let {getByRole, getAllByRole, queryByRole, rerender} = render(
+      <Provider theme={theme}>
+        <PendingAlert pendingAction="primary" />
+      </Provider>
+    );
+
+    expect(queryByRole('dialog')).toBeNull();
+    let triggerButton = getByRole('button');
+    await user.click(triggerButton);
+    act(() => {
+      jest.runAllTimers();
+    });
+
+    let dialog = getByRole('alertdialog');
+    expect(document.activeElement).toBe(dialog);
+
+    let buttons = getAllByRole('button');
+    expect(buttons).toHaveLength(3);
+
+    let pendingButton = buttons[2];
+    expect(pendingButton).toHaveTextContent('Accept');
+    expect(pendingButton).toHaveAttribute('aria-disabled', 'true');
+    let spinner = within(pendingButton).queryByRole('progressbar', {hidden: true});
+    expect(spinner).toBeVisible();
+
+    await user.click(buttons[0]);
```

Comment:
I think my previous comment would change this test, this shouldn't close/cancel the dialog until the pending state is resolved?
otherwise this needs to cancel/ignore the request results